liquidctl/driver/nvidia.py

```diff
@@ -238,32 +230,34 @@ def probe(cls, smbus, vendor=None, product=None, address=None, match=None,
                     or not smbus.description.startswith('NVIDIA i2c adapter 1 '):
                 continue
 
-            if not check_unsafe('smbus', 'rog_turing', **kwargs):
-                _LOGGER.warning("%s: assuming driver found at address %02x "
-                            "'smbus,rog_turing'",  desc, 0x2a)
-
+            selected_address = None
+
+            if check_unsafe('smbus', 'rog_turing', **kwargs):
+                for address in ADDRESSES:
+                    val1 = 0
+                    val2 = 0
+
+                    smbus.open()
+                    try:
+                        val1 = smbus.read_byte_data(address, 0x20)
+                        val2 = smbus.read_byte_data(address, 0x21)
+                    except:
+                        pass
+                    smbus.close()
+
+                    if val1 << 8 | val2 == ASUS_GPU_MAGIC_VALUE:
+                        selected_address = address
+                        break
+            else:
+                selected_address = cls._SENTINEL_ADDRESS
```

Comment:
What is the reason for `_SENTINEL_ADDRESS`? I get that it is a default, but not sure why it has that value, and does that name have any significance?

Reply:
I described it in the commit message: 7899412.

TL; DR. To prevent a card opened without `unsafe` to later be used with `unsafe`, and thus access an address we didn't actually validate.

The sentinel is checked for in `get_status` and `set_color`, and it also has a large value so that even if we forget, the lower level APIs will still probably refuse it.